test/test_basic_logic.py

```diff
@@ -1576,3 +1577,10 @@ def test_receiving_goaway_frame_with_unknown_error(self, frame_factory):
         assert c.state_machine.state == h2.connection.ConnectionState.CLOSED
 
         assert not c.data_to_send()
+
+    def test_stream_str(self):
```

Comment:
Nitpick: we’re really testing the `repr()`, and relying on the fallback that calling `str()` will call `repr()` if `__str__` isn’t defined. Could you change the test to say `repr` instead of `str`?